tasks/main.yml

```diff
@@ -65,29 +69,51 @@
   when: ansible_os_family == "Debian" and master_token is defined and legacy_gpg is undefined
 
 - name: "Adding packagecloud.io repository: {{ repository }}"
-  shell: curl -s "{{ debian_config_file_url_without_token }}" > {{ debian_config_file_location }} creates={{ debian_config_file_location }}
+  get_url:
+      url: "{{ debian_config_file_url }}"
+      dest: "{{ debian_config_file_location }}"
   register: added_deb_repository
   when: ansible_os_family == "Debian" and master_token is undefined
 
 - name: "Adding packagecloud.io repository: {{ repository }} with generated read token"
-  shell: curl -s "{{ debian_config_file_url_with_token }}" > {{ debian_config_file_location }} creates={{ debian_config_file_location }}
-  register: added_deb_repository
+  get_url:
+      url: "{{ debian_config_file_url }}"
+      dest: "{{ debian_config_file_location }}"
+      url_username: "{{ master_token }}"
+      force_basic_auth: yes
+  register: added_deb_repository_with_token
   when: ansible_os_family == "Debian" and master_token is defined
 
 - name: "Adding packagecloud.io repository: {{ repository }}"
-  shell: curl -s "{{ redhat_config_file_url_without_token }}" > {{ redhat_config_file_location }} creates={{ redhat_config_file_location }}
+  get_url:
+      url: "{{ redhat_config_file_url }}"
+      dest: "{{ redhat_config_file_location }}"
   register: added_rpm_repository
   when: ansible_os_family == "RedHat" and master_token is undefined
 
 - name: "Adding packagecloud.io repository: {{ repository }} with generated read token"
-  shell: curl -s "{{ redhat_config_file_url_with_token }}" > {{ redhat_config_file_location }} creates={{ redhat_config_file_location }}
-  register: added_rpm_repository
+  get_url:
+      url: "{{ redhat_config_file_url }}"
+      dest: "{{ redhat_config_file_location }}"
+      url_username: "{{ master_token }}"
+      force_basic_auth: yes
+  register: added_rpm_repository_with_token
   when: ansible_os_family == "RedHat" and master_token is defined
 
+- name: Disable repo_gpgcheck !workaround!
```

Comment:
Can you explain why this workaround was added?

Reply:
Saw the [commit message associated with this](https://github.com/computology/packagecloud-ansible-role/pull/4/commits/444080efe99bd40001866c8e5b04ddb2dbe7542e), that issue is not related to ansible at all so I removed it. That error was likely related to our recent GPG key migration https://blog.packagecloud.io/eng/2018/10/17/gpg-key-migration/

Reply:
I can confirm my playbooks work without this workarond. The commit was from over a year ago, so I assume Packagecloud fixed it by now.